ray_lightning/ray_ddp.py

```diff
@@ -274,87 +278,140 @@ def shutdown_remote():
 
     # All methods below are only executed in remote Ray workers.
 
+    def set_world_ranks(self, process_idx: int = 0):
+        """Set the appropriate rank attributes for the trainer."""
+        if self._is_remote:
+            self._global_rank = process_idx
+            self._local_rank, self._node_rank = self.global_to_local[
+                self.global_rank]
+
+    def _worker_setup(self, process_idx: int):
+        reset_seed()
+        self.set_world_ranks(process_idx)
+        rank_zero_only.rank = self.global_rank
+
+        # Taken from pytorch_lightning.utilities.distributed
+        # .init_dist_connection
+        # Modified to not use cluster environment.
+        if torch.distributed.is_available(
+        ) and not torch.distributed.is_initialized():
+            log.info(
+                f"initializing distributed: GLOBAL_RANK: {self.global_rank}, "
+                f"MEMBER: {self.global_rank + 1}/{self.world_size}")
+            torch.distributed.init_process_group(
+                self.torch_distributed_backend,
+                rank=self.global_rank,
+                world_size=self.world_size)
+
+            # on rank=0 let everyone know training is starting
+            rank_zero_info(
+                f"{'-' * 100}\n"
+                f"distributed_backend={self.torch_distributed_backend}\n"
+                f"All distributed processes registered. Starting with "
+                f"{self.world_size} processes\n"
+                f"{'-' * 100}\n")
+
     def execute_remote(self,
                        model: LightningModule,
                        global_rank: int,
                        queue: Queue = None):
-        """Train/test/eval function to be executed on each remote worker."""
+        """Train/test/eval function to be executed on each remote worker.
+
+        Modified from DDPSpawn._wrapped_function and DDPSpawn.new_process
```

Comment:
May be due to my unfamiliarity to the codebase/execution flow, but it seems like our implementation pushes a lot of logic directly into `execute_remote`/`execution_loop`. Is this intentional?

Reply:
Yes this is intentional- I think it's easier for maintainability to keep everything in 1 function rather than try to match the superclass exactly because the method signatures and usage in the superclass has changed quite a bit from last release.